pkg/node/manager/manager.go

```diff
@@ -362,16 +362,6 @@ func (m *manager) backgroundSync(ctx context.Context) error {
 // legacyNodeIpBehavior returns true if the agent is still running in legacy
 // mode regarding node IPs
 func (m *manager) legacyNodeIpBehavior() bool {
-	// Cilium < 1.7 only exposed the Cilium internalIP to the ipcache
-	// unless encryption was enabled. This meant that for the majority of
-	// node IPs, CIDR policy rules would apply. With the introduction of
-	// remote-node identities, all node IPs were suddenly added to the
-	// ipcache. This resulted in a behavioral change. New deployments will
-	// provide this behavior out of the gate, existing deployments will
-	// have to opt into this by enabling remote-node identities.
-	if m.conf.RemoteNodeIdentitiesEnabled() {
```

Comment:
This looks incomplete. `m.conf.RemoteNodeIdentitiesEnabled()` reads (at this point) non-existant value. Consider moving this commit to the earlier in the git history and make sure to also remove the `RemoteNodeIdentitiesEnabled()` function itself.